File: GuestPass.Api/models/Guest.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace GuestPass.Api.Models;

[Table("guests")]
public class Guest
{
    [Key]
    [Column("id")]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public Guid Id { get; set; }

    [Column("eventid")]
    public Guid? EventId { get; set; }

    [Column("name")]
    public string Name { get; set; } = string.Empty;

    [Column("email")]
    public string Email { get; set; } = string.Empty;

    [Column("qrcodetoken")]
    public string QRCodeToken { get; set; } = string.Empty;

    [Column("ischeckedin")]
    public bool IsCheckedIn { get; set; } = false;

    [Column("checkedinat")]
    public DateTimeOffset? CheckedInAt { get; set; }

    [Column("createdat")]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public DateTimeOffset CreatedAt { get; set; }

    // Navigation property
    [ForeignKey("EventId")]
    public Event? Event { get; set; }
}

File: guestpass-mobile/lib/data/models/guest.dart
/// Model Guest
class Guest {
  final String id;
  final String? eventId;
  final String name;
  final String email;
  final String qrCodeToken;
  final bool isCheckedIn;
  final DateTime? checkedInAt;
  final DateTime createdAt;

  Guest({
    required this.id,
    this.eventId,
    required this.name,
    required this.email,
    required this.qrCodeToken,
    required this.isCheckedIn,
    this.checkedInAt,
    required this.createdAt,
  });

  factory Guest.fromJson(Map<String, dynamic> json) {
    return Guest(
      id: json['id'] as String,
      eventId: json['eventId'] as String?,
      name: json['name'] as String,
      email: json['email'] as String,
      qrCodeToken: json['qrCodeToken'] as String? ?? json['qrcodeToken'] as String? ?? '',
      isCheckedIn: json['isCheckedIn'] as bool? ?? false,
      checkedInAt: json['checkedInAt'] != null
          ? DateTime.parse(json['checkedInAt'] as String)
          : null,
      createdAt: DateTime.parse(json['createdAt'] as String),
    );
  }
}
